src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -438,7 +438,13 @@ def _trim_elem(seg):
 
     @classmethod
     def _bracket_sensitive_look_ahead_match(
-        cls, segments, matchers, parse_context, start_bracket=None, end_bracket=None
+        cls,
+        segments,
+        matchers,
+        parse_context,
+        start_bracket=None,
+        end_bracket=None,
+        bracket_pairs_set="bracket_pairs",
```

Comment:
Added this parameter, which allows the available bracket pairs to vary per segment